apis/tempo/v1alpha1/ingress_types.go

```diff
@@ -0,0 +1,36 @@
+package v1alpha1
+
+type (
+	// IngressType represents how a service should be exposed (ingress vs route).
+	// +kubebuilder:validation:Enum=none;ingress;route
+	IngressType string
+)
+
+const (
+	// IngressTypeNone specifies that no ingress or route entry should be created.
+	IngressTypeNone IngressType = "none"
```

Comment:
Is none ever used? Seems its not set in the default webhook. 
```
IngressTypeUnknown IngressType = ""
```

Reply:
It's required for users who explicitly want to disable the ingress. If users set ingress to empty, the defaulter webhook will enable the ingress again.

Basically I need a way to distinguish between "unset" (the webhook should enable ingress by default) and "disabled" in case the user wants to disable the ingress (then the webhook must not re-enable the ingress).

Reply:
Alternatively we could decide to not enable the ingress by default. But I kind of like enabling it by default for UX reasons (I assume most people want it enabled).

Reply:
> Alternatively we could decide to not enable the ingress by default. But I kind of like enabling it by default for UX reasons (I assume most people want it enabled).

+1

Reply:
> > Alternatively we could decide to not enable the ingress by default. But I kind of like enabling it by default for UX reasons (I assume most people want it enabled).
> 
> +1

+1 for not enabling by default, or +1 for enabling by default? :)

Reply:
hm.. Not sure if i would like to expose something by default. ^^

Reply:
You're right, maybe that enabled-by-default ingress comes as a surprise for some users, which is bad for security. I've removed the code to enable it by default, and now we can use `""` as ingress-is-disabled value.